app/file.go

```diff
@@ -122,6 +122,15 @@ func (a *App) RemoveFile(path string) *model.AppError {
 	return backend.RemoveFile(path)
 }
 
+func (a *App) ListDirectory(path string) ([]string, *model.AppError) {
+	backend, err := a.FileBackend()
+	if err != nil {
+		return nil, err
+	}
+	paths, listErr := backend.ListDirectory(path)
```

Comment:
Is there a need to distinguish `listErr` from just reusing `err` here? (It's safe to write `paths, err := ...`, since Go will define `paths` as a new variable and just assign to the existing `err` variable.)